javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala

```diff
@@ -0,0 +1,43 @@
+package java.util.concurrent.atomic
+
+import scala.collection.mutable.ArrayBuffer
+
+class AtomicReferenceArray[E <: AnyRef](
+    length: Int) extends Serializable {
+
+  def this(array: Array[E]) = {
+    this(array.size)
+    inner = ArrayBuffer[E](array: _*)
+  }
+
+  private var inner: ArrayBuffer[E] = new ArrayBuffer[E](length)
```

Comment:
Ah, right. Use a `js.Array[E]`, then.
But then you have to initialize explicitly all elements to `null`:
```scala
private val inner: js.Array[E] = {
  val inner = new js.Array[E]()
  for (_ <- 0 until length)
    inner.push(null)
  inner
}
```